wumpus.py

```diff
@@ -103,7 +107,54 @@ def _breeze_stench_rules(self):
         return clauses
 
 
-def create_wumpus_world():
+def create_wumpus_world(map_width=4, map_height=4):
+    grid = [[set() for i in map_width] for j in map_height]
+    # can write a function for this type of random coord generation (nothing should be on 0,0)
+    wumpus_location = randrange(1, map_width * map_height)
+    wumpus_y = int(wumpus_location / map_width)
+    wumpus_x = wumpus_location - (wumpus_y * map_width)
+    grid[wumpus_y][wumpus_x].add(set([Property.WUMPUS]))
+    # can maybe use breeze stench rules to simplify this? or abstract into a function
+    if wumpus_x > 0:
+        if wumpus_y > 0:
+            grid[wumpus_y - 1][wumpus_x - 1].add([Property.STENCH])
+        if wumpus_y < map_height - 1:
+            grid[wumpus_y + 1][wumpus_x - 1].add([Property.STENCH])
+    if wumpus_x < map_width - 1:
+        if wumpus_y > 0:
+            grid[wumpus_y - 1][wumpus_x + 1].add([Property.STENCH])
+        if wumpus_y < map_height - 1:
+            grid[wumpus_y + 1][wumpus_x + 1].add([Property.STENCH])
+
+    max_traps_threshold = int(map_width * map_height * MAX_TRAPS_RATIO)
+    trap_locations = []
+    for i in range (map_width * map_height):
+        y = int(i / map_width)
+        x = i - (y * map_width)
+        if random(1) > TRAPS_GOLD_INCIDENCE_RATE: # tied to same rate for simplicity
```

Comment:
If you decide to implement a class, the `TRAPS_GOLD_INCIDENCE_RATE` and the `MAX_TRAPS_RATIO` could be properties, and those constants could be the default values